account-kit/rn-signer/src/client.ts

```diff
@@ -1,40 +1,75 @@
 /* eslint-disable import/extensions */
 import "./utils/mmkv-localstorage-polyfill";
+import "./utils/buffer-polyfill";
 import { type ConnectionConfig } from "@aa-sdk/core";
 import {
   BaseSignerClient,
+  OauthFailedError,
   type AlchemySignerClientEvents,
   type AuthenticatingEventMetadata,
   type CreateAccountParams,
   type EmailAuthParams,
   type GetWebAuthnAttestationResult,
+  type KnownAuthProvider,
   type OauthConfig,
   type OauthParams,
   type OtpParams,
   type SignupResponse,
   type User,
 } from "@account-kit/signer";
 import NativeTEKStamper from "./NativeTEKStamper";
+import {
+  OAuthProvidersError,
+  getOauthNonce,
+  getDefaultScopeAndClaims,
+} from "@account-kit/signer";
 import { z } from "zod";
+import { base64UrlEncode } from "./utils/base64UrlEncode";
+import { InAppBrowser } from "react-native-inappbrowser-reborn";
+import { parseSearchParams } from "./utils/parseUrlParams";
+
+type OauthState = {
+  authProviderId: string;
+  isCustomProvider?: boolean;
+  requestKey: string;
+  turnkeyPublicKey: string;
+  expirationSeconds?: number;
+  redirectUrl?: string;
+  openerOrigin?: string;
+};
 
 export const RNSignerClientParamsSchema = z.object({
   connection: z.custom<ConnectionConfig>(),
   rootOrgId: z.string().optional(),
+  oauthCallbackUrl: z
+    .string()
+    .optional()
+    .default("https://signer.alchemy.com/callback"),
 });
 
 export type RNSignerClientParams = z.input<typeof RNSignerClientParamsSchema>;
 
 // TODO: need to emit events
 export class RNSignerClient extends BaseSignerClient<undefined> {
   private stamper = NativeTEKStamper;
+  oauthCallbackUrl: string;
+  private validAuthenticatingTypes: AuthenticatingEventMetadata["type"][] = [
+    "email",
+    "otp",
+    "oauth",
+  ];
+
   constructor(params: RNSignerClientParams) {
-    const { connection, rootOrgId } = RNSignerClientParamsSchema.parse(params);
+    const { connection, rootOrgId, oauthCallbackUrl } =
+      RNSignerClientParamsSchema.parse(params);
 
     super({
       stamper: NativeTEKStamper,
       rootOrgId: rootOrgId ?? "24c1acf5-810f-41e0-a503-d5d13fa8e830",
       connection,
     });
+
+    this.oauthCallbackUrl = oauthCallbackUrl;
```

Comment:
how is this used? This is generally configured on the server, so I'm curious how this works here.

Reply:
nevermind, found it!